api/src/main/java/org/ehcache/config/Eviction.java

```diff
@@ -35,6 +35,7 @@ public boolean adviseAgainstEviction(Object key, Object value) {
    * @param <V> the value type for the advisor
    * @return an advisor where no mappings are advised against eviction
    */
+  @SuppressWarnings("unchecked")
```

Comment:
Tried it - does not work. We rely on specific generics bounds in some places. Think Store have `K,V` generics but have internal maps for `K, ValueHolder<V>` and in that context the generic bounds still apply only on `K` and `V`.